### DISCUSSION TITLE
Argument of type InstrumentedAttribute[T] cannot be assigned to parameter "X" of type T

### DISCUSSION THREAD

**Discussion by einarwar** [microsoft/pyright#6342] [Q&A] [answered]
When upgrading from pyright 1.1.333 to 1.1.334 there are several `reportGeneralTypeIssues` violations raised in code related to sqlalchemy models.

Using SQLAlchemy 2.0.23 with this minimal example: 
```python

from sqlalchemy.orm import declarative_base, Mapped, mapped_column

class BaseUnmapped:
    __allow_unmapped__ = True


Base = declarative_base(cls=BaseUnmapped)

class MyModel(Base):
    id: Mapped[int] = mapped_column(primary_key=True, index=True)
 
class Bar:
    def __init__(self, x: int):
        self.x = x

model = MyModel()
b = Bar(x=model.id)
```
This gives the following error in pyright 1.1.334:

`error: Argument of type "InstrumentedAttribute[int]" cannot be assigned to parameter "x" of type "int" in function "__init__"
    "InstrumentedAttribute[int]" is incompatible with "int" (reportGeneralTypeIssues)`

**Comment by erictraut** [comment, marked as answer]
This is the same issue as https://github.com/microsoft/pyright/issues/6325.

I didn't realize this behavior changed between 1.1.333 and 1.1.334. (This was mentioned in the other issue, but I missed it.) I'm going to reopen the other issue and take another look at it.
